tests/unit/objects/test_templates.py

```diff
@@ -0,0 +1,89 @@
+"""
+Gitlab API:
+https://docs.gitlab.com/ce/api/templates/dockerfiles.html
+https://docs.gitlab.com/ce/api/templates/gitignores.html
+https://docs.gitlab.com/ce/api/templates/gitlab_ci_ymls.html
+https://docs.gitlab.com/ce/api/templates/licenses.html
+https://docs.gitlab.com/ce/api/project_templates.html
+"""
+
+import pytest
+import responses
+
+from gitlab.v4.objects import (
+    Dockerfile,
+    Gitignore,
+    Gitlabciyml,
+    License,
+    ProjectDockerfileTemplate,
+    ProjectGitignoreTemplate,
+    ProjectGitlabciymlTemplate,
+    ProjectIssueTemplate,
+    ProjectLicenseTemplate,
+    ProjectMergeRequestTemplate,
+)
+
+
+@pytest.mark.parametrize(
+    "fixture, tmpl, tmpl_path, tmpl_mgr",
+    [
+        ("gl", Dockerfile, "dockerfiles", "dockerfiles"),
+        ("gl", Gitignore, "gitignores", "gitignores"),
+        ("gl", Gitlabciyml, "gitlab_ci_ymls", "gitlabciymls"),
+        ("gl", License, "licenses", "licenses"),
+        ("project", ProjectDockerfileTemplate, "dockerfiles", "dockerfile_templates"),
+        ("project", ProjectGitignoreTemplate, "gitignores", "gitignore_templates"),
+        (
+            "project",
+            ProjectGitlabciymlTemplate,
+            "gitlab_ci_ymls",
+            "gitlabciyml_templates",
+        ),
+        ("project", ProjectLicenseTemplate, "licenses", "license_templates"),
+        ("project", ProjectIssueTemplate, "issues", "issue_templates"),
+        (
+            "project",
+            ProjectMergeRequestTemplate,
+            "merge_requests",
+            "mergerequest_templates",
+        ),
+    ],
+    ids=[
+        "dockerfile",
+        "gitignore",
+        "gitlabciyml",
+        "license",
+        "project_dockerfile",
+        "project_gitignore",
+        "project_gitlabciyml",
+        "project_license",
+        "project_issue",
+        "project_mergerequest",
+    ],
+)
+def test_get_template(request, fixture, tmpl, tmpl_path, tmpl_mgr):
+    # Get the appropriate fixture and path. `owner` is the object that
+    # aggregates the template managers, i.e., Gitlab, Project
+    owner = request.getfixturevalue(fixture)
+    owner_path = "/" if fixture == "gl" else f"/projects/{owner.id}/"
```

Comment:
Thanks a lot for adding all the test cases here :bow: These days I lean more towards test repetition over complex if it means adding logic to the test cases.

I'd maybe split this into `test_get_template` and `test_get_project_template` to keep the parametrization a bit flatter. I realize we still need separate logic for the license ids. Only if you have the time, otherwise I'm happy go ahead with this, just happy to have the tests! :+1:

Reply:
Done.

Reply:
Note that I did an interactive rebase and performed the edits at each applicable commits--and force-pushed my changes. But other than that, I've implemented your changes :)